ch_backup/logic/access.py

```diff
@@ -28,13 +33,39 @@ def backup(self, context: BackupContext) -> None:
         """
         Backup access control entities.
         """
-        objects = context.ch_ctl.get_access_control_objects()
-        context.backup_meta.set_access_control(objects)
-        access_control = context.backup_meta.access_control
 
-        if self._has_replicated_access(context):
-            self._backup_replicated(access_control.acl_ids, context)
-        self._backup_local(access_control.acl_ids, context)
+        clickhouse_access_path = context.ch_ctl_conf["access_control_path"]
+        backup_tmp_path = os.path.join(
+            context.ch_ctl_conf["tmp_path"], context.backup_meta.name
+        )
+        user = context.ch_ctl_conf["user"]
+        group = context.ch_ctl_conf["group"]
+
+        create_dir(clickhouse_access_path, user, group)
+        with temporary_directory(backup_tmp_path, user, group):
+            objects = context.ch_ctl.get_access_control_objects()
+            context.backup_meta.set_access_control(objects)
+            access_control = context.backup_meta.access_control
+
+            if self._has_replicated_access(context):
+                self._backup_replicated(
+                    backup_tmp_path, access_control.acl_ids, context
+                )
+            self._backup_local(clickhouse_access_path, backup_tmp_path)
```

Comment:
hm, am I right that now `_backup_replicated()` just copies data from ZK to tmp dir, while `_backup_local()` copies local files from access path to tmp and that's it?

And so:
- `_backup_local` will always replace (in case of replicated storage) actual data from ZK with local version (because currently we do not clean them up)
- we still will have a local (and soon "outdated") version of ACL in access folder anyway?

Reply:
> Yes, we will have it. I think make sense to cleanup /var/lib/clickhouse/access in restore operation.

because I thought that originally we could have potential issue with local copies after each backup - in case of troubles with ZK we can't use replicated storage - so we can have a conflict with local sql files

so maybe we should clean it up after backup operation as well?

Reply:
>so maybe we should clean it up after backup operation as well?

Im not sure because suppose:
1) we have 2 acl entries: 1 - old one without zk, 2 - with zk
2) perform backup -> removing from `/var/lib/clickhouse/acces`.
3) So after  ch-server restart we will lose 1 entry.

It seems fine to do cleanup on restore, but we will lose non-replicated records when we do a backup.

Reply:
> we have 2 acl entries: 1 - old one without zk, 2 - with zk

is it already possible to have 2 entities in different storages (local and replicated)? because I've tested it initially (22.x), and found that you can use only one storage type for that (if already no - maybe it makes sense)

and if 1 possible storage is still relevant - it's better to clean it up after a backup as well (think about your current state on the clusters - they all have local files), or move/remove them during migration from replicated to local

otherwise I think current approach works only for new clusters and with new ch-backup version